src/toil/test/sort/sort.py

```diff
@@ -158,32 +158,53 @@ def getMidPoint(file, fileStart, fileEnd):
+            fileHandle.write(line)
 
-    parser.add_argument("--fileToSort", dest="fileToSort",
-                      help="The file you wish to sort")
 
-    parser.add_argument("--N", dest="N",
-                      help="The threshold below which a serial sort function is"
-                      "used to sort file. All lines must of length less than or equal to N or program will fail",
-                      default=10000)
+def main(options=None):
+    if not options:
+        parser = ArgumentParser()
+        Job.Runner.addToilOptions(parser)
+        parser.add_argument('--numLines', default=defaultLines, help='Number of lines in file to sort.', type=int)
+        parser.add_argument('--lineLength', default=defaultLineLen, help='Length of lines in file to sort.', type=int)
+        parser.add_argument("--fileToSort", dest="fileToSort", help="The file you wish to sort")
+        parser.add_argument("--N", dest="N",
+                            help="The threshold below which a serial sort function is used to sort file. "
+                                 "All lines must of length less than or equal to N or program will fail",
+                            default=10000)
+        parser.add_argument('--downCheckpoints', action='store_true',
+                            help='If this option is set, the workflow will make checkpoints on its way through'
+                                 'the recursive "down" part of the sort')
+        options = parser.parse_args()
 
-    options = parser.parse_args()
+    fileName = options.fileToSort
 
     if options.fileToSort is None:
-        raise RuntimeError("No file to sort given")
-
-    if not os.path.exists(options.fileToSort):
-        raise RuntimeError("File to sort does not exist: %s" % options.fileToSort)
+        # make the file ourselves
+        fileName = 'fileToSort.txt'
+        print 'No sort file specified. Generating one automatically called %s.' % fileName
+        makeFileToSort(fileName=fileName, lines=options.numLines, lineLen=options.lineLength)
+    else:
+        if not os.path.exists(options.fileToSort):
+            raise RuntimeError("File to sort does not exist: %s" % options.fileToSort)
 
     if int(options.N) <= 0:
         raise RuntimeError("Invalid value of N: %s" % options.N)
 
-    #Now we are ready to run
-    Job.Runner.startToil(Job.wrapJobFn(setup, options.fileToSort, int(options.N), False,
-                                       memory=sortMemory), options)
+    # Now we are ready to run
+    with Toil(options) as toil:
+        if not toil.options.restart:
+            sortFileURL = 'file://' + os.path.abspath(fileName)
+            sortFileID = toil.importFile(sortFileURL)
+            sortedFileID = toil.start(Job.wrapJobFn(setup, sortFileID, int(options.N), options.downCheckpoints,
+                                                    memory=sortMemory))
+            toil.exportFile(sortedFileID, sortFileURL)
```

Comment:
I think this is what's causing your test failures. The file isn't being exported on restart, so it isn't being overwritten properly if the workflow totally fails once.

Moving this line to line 208/207 and moving it up one level of indentation (so it gets run in both code paths) should fix things.